### PR TITLE
docs(self-managed): add ingress setup guide

### PR DESCRIPTION
**Pull request by aabouzaid**
## What is the purpose of the change

Adding a guide for Ingress setup with single and multi domains. 

Please note that this guide depends on PR no. #1241 and I just put it in a separate PR to make it easy to review.

## Are there related marketing activities

No.

## When should this change go live?

ASAP.

## PR Checklist

- [x] My changes are not for an already released version.
- [x] My changes apply to future versions, and I have added them to the main `/docs` directory.
- [x] My changes do not require an Engineering review.
- [x] My changes require a [technical writer review](https://github.com/camunda/camunda-platform-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned @christinaausley as a reviewer, or my changes do not require a technical writer review.

### CODE REVIEW

File: docs/self-managed/platform-deployment/ingress-setup.md [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+
+There are no significant differences between the two setups. Rather, they both offer flexibility for different workflows.
+
+:::note
+Camunda Platform Helm chart doesn't manage or deploy Ingress controllers, it only deploys Ingress objects. Hence, this Ingress setup will not work without Ingress controller running in your cluster.
+:::
+
+## Preparation
+
+- An Ingress controller should be deployed in advance. The examples below use `Nginx` Ingress controller, but any Ingress controller could be used by setting `ingress.className`.
```
Comment (christinaausley) [inline docs/self-managed/platform-deployment/ingress-setup.md:20]: I adjusted to "The examples below use `Nginx` Ingress controller" -- is this what you meant?

File: docs/self-managed/platform-deployment/ingress-setup.md [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+
+Once deployed, you can access the Camunda Platform components on:
+
+- **Web applications:** `https://camunda.example.com/[identity|operate|optimize|tasklist]`
+- **Keycloak authentication:** `https://camunda.example.com/auth`
+- **Zeebe Gateway:** `grpc://zeebe.camunda.example.com`
+
+## Separated Ingress setup
+
+In this setup, each Camunda Platform component has its own Ingress/domain. There is no need to set the context (unless there is a good reason for that) since `/` is used as a default base. Here, we just need to set the Ingress configuration and authentication redirect URLs.
```
Comment (christinaausley) [inline docs/self-managed/platform-deployment/ingress-setup.md:85]: "(unless there is a good reason for that)" -- can you clarify here? Are there particular instances or scenarios where there might be a good reason? If there aren't any worth mentioning, I would remove this.

**Review by christinaausley — CHANGES_REQUESTED** [review]
A few comments for clarification and technical adjustments in latest commit.

Additionally, when you say "Camunda Platform" throughout this doc, could you specify? For example, "Camunda Platform 8" or "Camunda Platform 7" to avoid any confusion.

File: docs/self-managed/platform-deployment/ingress-setup.md [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+    host: "tasklist.camunda.example.com"
+
+zeebe-gateway:
+  ingress:
+    enabled: true
+    className: nginx
+    host: "zeebe.camunda.example.com"
+```
+
+Using the custom values file, [deploy Camunda Platform as usual](../deployment.md):
```
Comment (christinaausley) [inline docs/self-managed/platform-deployment/ingress-setup.md:150]: This link is failing -- do you mean `platform-8-deployment.md`?

File: docs/self-managed/platform-deployment/ingress-setup.md (hunk 1 above)
Comment (aabouzaid) [reply docs/self-managed/platform-deployment/ingress-setup.md:20]: yes exactly :+1:

File: docs/self-managed/platform-deployment/ingress-setup.md (hunk 2 above)
Comment (aabouzaid) [reply docs/self-managed/platform-deployment/ingress-setup.md:85]: yeah if the domain will be used for multiple resources.
but I'd say who does that will know what to do ... so I will remove it.

File: docs/self-managed/platform-deployment/ingress-setup.md (hunk 3 above)
Comment (aabouzaid) [reply docs/self-managed/platform-deployment/ingress-setup.md:150]: this guide uses the new structure from #1241
I will move it there once the review is done

**Comment by aabouzaid** [comment]
> A few comments for clarification and technical adjustments in latest commit.
> 
> Additionally, when you say "Camunda Platform" throughout this doc, could you specify? For example, "Camunda Platform 8" or "Camunda Platform 7" to avoid any confusion.

Do we need to specify in the doc body?
i.e. if the page title and the first paragraph on the page mention Camunda Platform 8, do we really need to add the version in each paragraph?

Personally, it sounds redundant, but if we have a convention for that, I will change it.

**Comment by christinaausley** [comment]
> > A few comments for clarification and technical adjustments in latest commit.
> > Additionally, when you say "Camunda Platform" throughout this doc, could you specify? For example, "Camunda Platform 8" or "Camunda Platform 7" to avoid any confusion.
> 
> Do we need to specify in the doc body? i.e. if the page title and the first paragraph on the page mention Camunda Platform 8, do we really need to add the version in each paragraph?
> 
> Personally, it sounds redundant, but if we have a convention for that, I will change it.

Totally understandable. Alongside the [Product Naming page in Confluence](https://confluence.camunda.com/pages/viewpage.action?spaceKey=HAN&title=Product+Names), we do indeed refer to product names in full. This helps avoid any confusion users might have between products, as sometimes we still get questions about which product we're referring to. This could change in the future, but given the semi-recent product release, we do still prefer to refer to names in full for clarity.

**Comment by aabouzaid** [comment]
@christinaausley done :heavy_check_mark:

**Review by christinaausley — APPROVED** [review]

**Review by akeller — APPROVED** [review]
I had to fix a couple of links that weren't working on `build` but this is good to go now!

**Pull request merged**